lightningd/peer_htlcs.c

```diff
@@ -431,6 +445,12 @@ enum onion_type send_htlc_out(struct channel *out, u64 amount, u32 cltv,
 			    payment_hash, onion_routing_packet, in);
 	tal_add_destructor(hout, destroy_hout_subd_died);
 
+	/* We give it 30 seconds to commit htlc. */
+	if (!out->htlc_timeout)
```

Comment:
How can it happen that we ever attempt to add it twice?

Reply:
I'm pretty sure you're right: I will switch this to an assert().

Reply:
(I started fixing this then realized that the timer is on the channel, not the HTLC).

I replaced comment with:

```
	/* Give channel 30 seconds to commit (first) htlc. */
```

Reply:
Thanks for clarifying